api/domain_handlers_test.go

```diff
@@ -24,3 +34,89 @@ func TestNormalizeDomain(t *testing.T) {
 		}
 	}
 }
+
+// TestIneligible tests that IneligibleDomainState Database is populated when the Ineligible endpoint is called.
+func TestIneligible(t *testing.T) {
+	api, _, mockHstspreload, mockPreloadlist := mockAPI(0 * time.Second)
+
+	// database.Scan values for testing
+
+	emptyScan := database.Scan{
+		Issues: []hstspreload.Issues{emptyIssues},
+	}
+
+	warningScan := database.Scan{
+		Issues: []hstspreload.Issues{issuesWithWarnings},
+	}
+
+	errorScan := database.Scan{
+		Issues: []hstspreload.Issues{issuesWithErrors},
+	}
+
+	TestEligibleResponses := map[string]hstspreload.Issues{
+		"garron.net":   emptyIssues,
+		"badssl.com":   issuesWithWarnings,
+		"chromium.org": emptyIssues,
+		"godoc.og":     issuesWithErrors,
+		"dev":          issuesWithWarnings,
+		"example.com":  issuesWithErrors,
+	}
+	TestPreloadlist := preloadlist.PreloadList{Entries: []preloadlist.Entry{
+		{Name: "garron.net", Mode: preloadlist.ForceHTTPS, IncludeSubDomains: true, Policy: preloadlist.Bulk18Weeks},
+		{Name: "chromium.org", Mode: preloadlist.ForceHTTPS, IncludeSubDomains: false, Policy: preloadlist.Bulk1Year},
+		{Name: "godoc.og", Mode: "", IncludeSubDomains: true, Policy: preloadlist.Bulk18Weeks},
```

Comment:
Good idea to test something with a mode other than "force-https".

The preload list (transport_security_state_static.json) has some entries that aren't for preloading HSTS. Only the entries with a mode of "force-https" are preloading HSTS. (The other entries are for preloading key pinning.) This project only cares about entries preloaded for HSTS, so for entries (like this one) that don't have a mode of "force-https", nothing should be done. The current behavior is that a scan is still done.